src/main/java/org/joshsim/engine/simulation/Replicate.java

```diff
@@ -33,36 +54,75 @@ public interface Replicate {
    * @param stepNumber the number of the step to have current state saved as.
    * @throws IllegalArgumentException if a TimeStep of the given step number already exists.
    */
-  void saveTimestep(long stepNumber);
+  public void saveTimeStep(long stepNumber) {
+    if (pastTimeSteps.containsKey(stepNumber)) {
+      throw new IllegalArgumentException("TimeStep already exists for step number " + stepNumber);
+    }
+    HashMap<GeoKey, Entity> frozenPatches = new HashMap<>();
+    for (Patch patch : presentTimeStep.values()) {
+      frozenPatches.put(patch.getKey().orElseThrow(), patch.freeze());
```

Comment:
Perfect - this is what I expect. I will take care of ensuring that cascades downwards through EngineValue! I think this is all that Replicate needs to do.